core/src/main/java/jeeves/server/context/ServiceContext.java

```diff
@@ -246,12 +351,24 @@ public void setAsThreadLocal() {
         }
         ApplicationContextHolder.set(this.getApplicationContext());
         // step three detail on present re-allocation
-        allocation = new Throwable("ServiceContext allocated to thread");
+        allocation = new Throwable("ServiceContext "+_service+" allocated to thread");
 
         unexpected += "\n\tService '"+_service+"' allocate: " + allocation.getStackTrace()[1];
         checkUnexpectedState( unexpected );
     }
 
+    /**
+     * Check if _service name is available, or raise a NullPointerException if clear() has already been called.
+     *
+     * @param message message to use if clear() has already been called.
+     */
+    protected void checkCleared(String message){
+        if( _service == null){
+            String unavailable = message + " - service context no longer available\nCleared by " + deAllocation.getStackTrace()[1];
```

Comment:
This reports the line number of the code that called clear() making it easier to debug issues where another area of the code has accidentally cleared a service context you were using.